From fa14654bac8991e3d88dcc2aee14d2175395fc5b Mon Sep 17 00:00:00 2001
From: "Wu, Qiming" <qi-ming.wu@intel.com>
Date: Wed, 27 Jun 2018 03:58:55 +0200
Subject: [PATCH] Merge pull request #28 in SW_UGW/linux from
 feature/DRVLIB_SW-794-enable-top-level-pib_ctrl.wakeup_int_en-bit-during-boot-up
 to master

* commit '9fc002e7b5d3593d85ea4099e9d8694f913aa290':
  Enable PON wakeup intr, reset PON DQCNTR during init
---
 drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c b/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
index 1a24675614d1..8aa8683e4c07 100644
--- a/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
@@ -1928,7 +1928,11 @@ static s32 do_port_setting(u32 *pmac, u32 flags, u32 *cbm_port,
 		/*clear PIB bypass*/
 		ctrl.pib_bypass = 0;
 		ctrl.pib_en = 1;
-		config_pib_ctrl(&ctrl, FLAG_PIB_BYPASS | FLAG_PIB_ENABLE);
+		ctrl.wakeup_intr_en = 1;
+		config_pib_ctrl(&ctrl,
+				FLAG_PIB_BYPASS |
+				FLAG_PIB_ENABLE |
+				FLAG_PIB_WAKEUP_INTR);
 		if (cqm_ctrl->force_xpcs)
 			set_ifmux(FMX_WAN_AON_MODE);
 		else
@@ -3684,6 +3688,8 @@ static int cqm_falconmx_probe(struct platform_device *pdev)
 	}
 	for (i = 0; i < CQM_QID2EP_DW_NUM; i++)
 		cbm_w32((cqm_ctrl->qid2ep + i * 4), 0x0);
+	for (i = 0; i < FMX_MAX_PON_PORTS; i++)
+		cbm_w32((cqm_ctrl->pon_dqm_cntr + i * 4), 0x0);
 	spin_unlock_irqrestore(&cqm_qidt_lock, sys_flag);
 	bm_init(pdev);
 	qos_init(pdev);
